File: server/src/main/java/dev/dtos/requests/OrderRequest.java
package dev.dtos.requests;

import lombok.Data;
import lombok.NonNull;

import java.util.Date;
import java.util.List;

@Data
@NonNull
public class OrderRequest {
    private String status;
    private String shippingAddress;
    private Date createdAt;
    private Date updateAt;
    private String paymentMethod;
    private String shippingMethod;
    private Double total;
    private List<OrderItemsRequest> orderItems;
}

File: TEST/app/src/main/java/com/javaproject/myapplication/data/model/request/OrderRequest.kt
package com.javaproject.myapplication.data.model.request

import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime

data class OrderRequest (
    @SerializedName("status")
    val status: String,

    @SerializedName("shippingAddress")
    val shippingAddress: String,

    @SerializedName("createdAt")
    val createdAt: LocalDateTime,

    @SerializedName("updateAt")
    val updateAt: LocalDateTime,

    @SerializedName("paymentMethod")
    val paymentMethod: String,

    @SerializedName("shippingMethod")
    val shippingMethod: String,

    @SerializedName("total")
    val total: Double,

    @SerializedName("orderItems")
    val orderItems: List<OrderItem>
)